stl/inc/ranges

```diff
@@ -6842,6 +6843,472 @@ namespace ranges {
+            requires _All_forward<_IsConst, _ViewTypes...>
+        struct _Category_base<_IsConst> {
+            using iterator_category = input_iterator_tag;
+        };
+
+        template <bool _IsConst>
+        class _Iterator : public _Category_base<_IsConst> {
+        private:
+            /* [[no_unique_address]] */ tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current;
+
+            friend zip_view;
+
+            constexpr explicit _Iterator(tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...> _Current_) noexcept(
+                is_nothrow_move_constructible_v<
+                    tuple<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>...>>) // strengthened
+                : _Current(_STD move(_Current_)) {}
+
+        public:
+            using iterator_concept = conditional_t<_All_random_access<_IsConst, _ViewTypes...>,
+                random_access_iterator_tag,
+                conditional_t<_All_bidirectional<_IsConst, _ViewTypes...>, bidirectional_iterator_tag,
+                    conditional_t<_All_forward<_IsConst, _ViewTypes...>, forward_iterator_tag, input_iterator_tag>>>;
+
+            using value_type      = tuple<range_value_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+            using difference_type = common_type_t<range_difference_t<_Maybe_const<_IsConst, _ViewTypes>>...>;
+
+            _Iterator() = default;
+
+            constexpr _Iterator(_Iterator<!_IsConst> _Rhs) noexcept(
+                (is_nothrow_convertible_v<iterator_t<_ViewTypes>,
+                     iterator_t<_Maybe_const<!_IsConst, _ViewTypes>>> && ...)) // strengthened
+                // clang-format off
+                requires(_IsConst &&
+            (convertible_to<iterator_t<_ViewTypes>, iterator_t<_Maybe_const<_IsConst, _ViewTypes>>> && ...))
+                // clang-format on
+                : _Current(_STD move(_Rhs._Current)) {}
+
+            _NODISCARD constexpr auto operator*() const
+                noexcept((noexcept(*(_STD declval<iterator_t<_Maybe_const<_IsConst, _ViewTypes>>>()))
+                          && ...)) /* strengthened */ {
+                return _Tuple_transform([](auto& _Itr) -> decltype(auto) { return *_Itr; }, _Current);
+            }
+
+            constexpr _Iterator& operator++() noexcept(
+                noexcept(_Tuple_for_each([](auto& _Itr) { ++_Itr; }, _Current))) /* strengthened */ {
+                _Tuple_for_each([](auto& _Itr) { ++_Itr; }, _Current);
+                return *this;
+            }
+
+            constexpr void operator++(int) noexcept(noexcept(++(_STD declval<_Iterator>()))) /* strengthened */ {
+                ++*this;
+            }
+
+            constexpr _Iterator operator++(int) noexcept(
+                noexcept(++(_STD declval<_Iterator>()))
+                && is_nothrow_copy_constructible_v<_Iterator<_IsConst>>) // strengthened
+                requires _All_forward<_IsConst, _ViewTypes...> {
+                const auto _Temp = *this;

```

Comment:
Although `const` is one of our favorite keywords and we mark local variables with it whenever possible, this is a rare exception - when returning a local variable, this can inhibit automatic moving (if the Named Return Value Optimization doesn't activate for whatever reason). The Standardese depicts `auto tmp` and we should use non-const `auto` here to match it. Occurs below.